### PR TITLE
[mypyc] feat: support negative index in TupleGet op

### PR DESCRIPTION
**Pull request by BobTheBuidler**
This PR modifies `TupleGet.__init__` to automatically convert negative indexes to positive indexes instead of crashing at the assert

This won't change functionality on its own, since none of the existing calling locations can pass a negative value, but will allow us to pass negative values in https://github.com/python/mypy/pull/19972 so I think we should consider this PR a prerequisite to that one

### CODE REVIEW

**Comment by BobTheBuidler** [comment]
I won't add any tests to this PR but I'll ensure #19972 tests both the positive and negative case before we merge

File: mypyc/ir/ops.py [hunk 1]
```diff
@@ -1045,10 +1045,17 @@ class TupleGet(RegisterOp):
...
         self.src = src
         self.index = index
-        assert isinstance(src.type, RTuple), "TupleGet only operates on tuples"
-        assert index >= 0
+        if index < 0:
+            self.index += src_len
+        assert (
+            self.index <= src_len - 1
+        ), f"Index out of range.\nsource type: {src.type}\nindex: {index}"
         self.type = src.type.types[index]
```
Comment (p-sawicki) [inline mypyc/ir/ops.py:1059]: should this use `self.index` as an index now?

Comment (BobTheBuidler) [reply mypyc/ir/ops.py:1059]: from a correctness standpoint both values work fine, but the current variation should be marginally faster in a meaningless way. I have no real opinion for or against changing it

File: mypyc/ir/ops.py [hunk 2]
```diff
@@ -1045,10 +1045,17 @@ class TupleGet(RegisterOp):
 
     def __init__(self, src: Value, index: int, line: int = -1, *, borrow: bool = False) -> None:
         super().__init__(line)
+        assert isinstance(
+            src.type, RTuple
+        ), "TupleGet only operates on tuples, not {type(src.type).__name__}"
```
Comment (p-sawicki) [inline mypyc/ir/ops.py]:
```suggestion
        ), f"TupleGet only operates on tuples, not {type(src.type).__name__}"
```

File: mypyc/ir/ops.py (hunk 1 above)
Comment (p-sawicki) [reply mypyc/ir/ops.py:1059]: ah right it doesn't matter, seems fine to leave it.
